server/games/typedefs.py

```diff
@@ -36,21 +36,10 @@ class Victory(Enum):
 
 
 @unique
-class GameType(Enum):
-    COOP = 0
-    CUSTOM = 1
-    MATCHMAKER = 2
-
-    @staticmethod
-    def from_string(value: str) -> Optional["GameType"]:
-        """
-        # Returns
-        `GameType` or `None` if the string is not valid
-        """
-        return GameType.__members__.get(value.upper())
-
-    def to_string(self) -> str:
-        return self.name.lower()
+class GameType(str, Enum):
```

Comment:
Why does it inherent from `str`?

Reply:
without it does not have the ability to compare.

Reply:
Does one of the tests fail? Normally you compare enums using `is`. I’m sure we can fix our tests so that we don’t need to inherent from `str` here.

Reply:
FAILED tests/integration_tests/test_matchmaker.py::test_game_launch_message_game_options - concurrent.futures._base.TimeoutError
FAILED tests/integration_tests/test_matchmaker.py::test_game_matchmaking_timeout - concurrent.futures._base.TimeoutError
FAILED tests/integration_tests/test_matchmaker.py::test_anti_map_repetition - concurrent.futures._base.TimeoutError
FAILED tests/integration_tests/test_teammatchmaker.py::test_game_ratings - concurrent.futures._base.TimeoutError
FAILED tests/integration_tests/test_teammatchmaker.py::test_ratings_initialized_based_on_global_persisted - concurrent.futures._base.TimeoutError

these fail without the str inheritage
in contrast to: 
FAILED tests/integration_tests/test_matchmaker.py::test_game_launch_message_game_options - concurrent.futures._base.TimeoutError
FAILED tests/integration_tests/test_matchmaker.py::test_game_matchmaking_disconnect - concurrent.futures._base.TimeoutError
with the str inheritage

Reply:
i have the idea with str from here:
https://stackoverflow.com/questions/58608361/string-based-enum-in-python

Reply:
Can you figure out what exception is causing those tests to fail? I don’t think we should need to compare the enum to strings anywhere, that seems like an issue with the code to me rather than an issue with the enum definition.

Reply:
yeah ill hang on to it.

Reply:
should work now, seems like a rare error, not connected to my changes.